app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,7 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.get_ingredient_names
```

Comment:
You're adhering to MVC very well! In this case though, having @ingredients and @cost instance variables is not strictly necessary. While we don't want any AR logic in the view, it's okay to call @recipe.ingredient_names or @cost.ingredients_cost from the view and just pass the one @recipe variable to it. This is just for the future if you want to minimize instance variables. Just another way to do it!

Also, this seems like an appropriate comment to add a response to your overall PR description:
For getting ingredients for a given recipe through associations (i.e. `recipe.ingredients`), it is totally okay to call the `.ingredients` association method on a recipe in either the controller or the view because ingredients is just an attribute on a recipe, and specifically an attribute set up by associations. You're not doing any AR logic here so it's safe to go in model, view or controllers.